neos/neos-development-collection
# BUGFIX: Have FlowQuery operations reliably detect nodes during canEvaluate

When a collection of nodes is referenced some items may become invisible over time or may be deleted. In this case the references may end up with an array that has no key 0. Since this key was used to check wether a flowQuery operation can be applied this could lead to errors. 

The change adjusts the detection by looking at using reset to get the first value if the context is an array and by using an foreach that returns in the first cycle for traversables.

**Review Instructions** 

To test this you can use the contentReferences Element in Neos.Demo. 

1. Create 3 contents
2. Create a reference element that references all three
3. Hide the content that is selected first
4. Apply `@process.filterContent( q(value).filter('[instanceof Neos.Neos:Content]').get())` to that.

Without the change in live context no content is shown because the first item is not available live. 
This change allows to still proceed with the remaining nodes.

Fixes: #3624
